sns-client/src/main/java/com/github/eeriefoods/snsclient/controller/CourseTableController.java

```diff
@@ -0,0 +1,108 @@
+import com.github.eeriefoods.snsclient.model.Student;
+import com.github.eeriefoods.snsclient.service.CourseService;
+
+import javafx.fxml.FXML;
+
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.control.cell.TextFieldTableCell;
+import javafx.util.StringConverter;
+import javafx.util.converter.IntegerStringConverter;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class CourseTableController {
+    @FXML public TableView<Course> TCCView;
+    @FXML private TableColumn<Course, String> TCCId, TCCRoom;
+
+    @FXML private TableColumn<Course, Integer> TCCMemberCount;
+
+    private StudentToolBarController studentToolBarController;
+    private StudentTableController studentTableController;
+    private TableView<Student> studentTableView;
+
+    public void injectMainController(MainController mainController){
+        this.studentTableController = mainController.studentTableController;
+        this.studentTableView = mainController.studentTableController.TCS_View;
+        this.studentToolBarController = mainController.studentToolBarController;
+    }
+
+    @FXML private void initialize() {
+
+        initTableCellFactories();
+        initTableCellEdit();
+        updateCourseTable();
+        TCCView.setEditable(true);
+    }
+
+    private void initTableCellEdit() {
+        TCCId.setOnEditCommit(event -> {
+            event.getTableView().getItems().get(event.getTablePosition().getRow()).setId(event.getNewValue());
+            try {
+                updateCourse(event.getRowValue());
+            } catch (IOException | InterruptedException e) {
+                e.printStackTrace();
+            }
+        });
+
+        TCCRoom.setOnEditCommit(event -> {
+            event.getTableView().getItems().get(event.getTablePosition().getRow()).setRoom(event.getNewValue());
+            try {
+                updateCourse(event.getRowValue());
+            } catch (IOException | InterruptedException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    private void initTableCellFactories() {
+        TCCId.setCellValueFactory(new PropertyValueFactory<>("id"));
+        TCCId.setCellFactory(TextFieldTableCell.forTableColumn());
+        TCCRoom.setCellValueFactory(new PropertyValueFactory<>("room"));
+        TCCRoom.setCellFactory(TextFieldTableCell.forTableColumn());
+        TCCMemberCount.setCellFactory(TextFieldTableCell.forTableColumn(new IntegerStringConverter()));
+        TCCMemberCount.setCellValueFactory(new PropertyValueFactory<>("memberCount"));
+    }
+
+    public List<Course> loadCourseList() throws IOException, InterruptedException{
+            return CourseService.getCourses();
+    }
+
+    private void updateCourse(Course course) throws IOException, InterruptedException { //REMOVE EXCEPTIONS HERE!
```

Comment:
Move try catch to here?

Reply:
fixed with next commit
also: deprecated because errorhandling is handled by parent function, thus not necessary